src/actions/smartWalletActions.js

```diff
@@ -1678,3 +1679,15 @@ export const setSmartWalletEnsNameAction = (username: string) => {
     dispatch(insertTransactionAction(historyTx, accountId));
   };
 };
+
+export const initSmartWalletSdkWithPrivateKeyOrPinAction = ({ privateKey: _privateKey, pin }: InitSmartWalletProps) => {
+  return async (dispatch: Dispatch) => {
+    let privateKey = _privateKey;
+    if (!_privateKey && pin) {
+      privateKey = await getPrivateKeyFromPin(pin, dispatch);
+    } else {
+      return;
+    }
+    await dispatch(initSmartWalletSdkAction(privateKey));
```

Comment:
I don't get this part. So if privateKey is provided in the function argument, we don't want to dispatch `initSmartWalletSdkAction` and instead we just return null? Are you sure you want to keep this?:
```
else {
  return;
}
```
In this scenario `dispatch(initSmartWalletSdkAction(privateKey));` will only be invoked if the user authenticates with pin, and not if they authenticate with privateKey. Is this intended?

Reply:
Ahh, darn it, must have overlooked something. Thanks! Will fix it